swarm/network/stream/stream.go

```diff
@@ -764,6 +768,40 @@ func (r *Registry) createSpec() {
 	r.spec = spec
 }
 
+//An accountable message needs some meta information attached to it
+//in order to evaluate the correct price
+type StreamerPrices struct {
+	priceMatrix map[reflect.Type]*protocols.Price
+	registry    *Registry
+}
+
+//Price implements the accounting interface and returns the price for a specific message
+func (spo *StreamerPrices) Price(msg interface{}) *protocols.Price {
+	typ := reflect.TypeOf(msg).Elem()
+	return spo.priceMatrix[typ]
```

Comment:
I don't remember why we "optimized" it away with @zelig, it was there some time. Maybe because the assignment is just below?

Reply:
Now I remember partly: it was agreed with @zelig that the `Price` interface does not return an error, just a `*Price` pointer.

Now if in `return spo.priceMatrix[typ]` the type does not exist, it would return `<nil>`, which is expected on the caller's side `p2p/protocols/accounting.go`:
```
139   price := ah.Price(msg)
140   //this message doesn't need accounting
141   if price == nil {
142     return nil
143   }
```

Reply:
okidok